CHANGELOG.md

```diff
@@ -4,6 +4,14 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.44.0]
+
+### What's Changed
+
+* Rust toolchain upgraded to 2024-01-08 rust toolchain by @adpaco-aws, @celinval, @zhassan-aws
```

Comment:
```suggestion
* Rust toolchain upgraded to `nightly-2024-01-08` by @adpaco-aws @celinval @zhassan-aws
```
Feel free to leave in the commas, it's just that we haven't used them in other occasions